pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -510,15 +587,22 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(readyCondition, lastReadyCondition) {
```

Comment:
I didn't understand this. If we get here, we know LastProbeTime didn't change. So how could ready condition have changed (i.e. how can it ever be true that !api.Semantic.DeepEqual(readyCondition, lastReadyCondition) ?